packages/core/src/stream/sse-stream.ts

```diff
@@ -39,28 +39,35 @@ export class SseStream<Item> implements AsyncIterable<Item> {
             continue;
           }
 
+          let data: any;
+
           try {
-            const data = JSON.parse(sse.data);
-            if (data?.error) {
-              throw new Error(data.error);
-            }
-            // Yield also the event if it exists, otherwise just the data
-            yield sse.event === null
-              ? data
-              : ({ event: sse.event, data } as any);
+            data = JSON.parse(sse.data);
           } catch (e: any) {
             logger.error(`Could not parse message into JSON: ${sse.data}`);
             logger.error(`From chunk: ${sse.raw}`);
             throw e;
           }
+
+          if (data?.error) {
+            logger.error(
+              `Error received from server: ${JSON.stringify(data.error)}`
+            );
+            throw new Error(JSON.stringify(data.error));
```

Comment:
@emincihangeri I tried out and saw the problem. Here I now stringify the error message because a JS error normally consists of error.name and error.message, which both are suppose to be strings.